main.go

```diff
@@ -30,9 +44,15 @@ func UpdateTableNames(node sqlparser.Node, nameMapper func(string) (string, bool
 	if err := sqlparser.Walk(func(node sqlparser.Node) (bool, error) {
 		if table, ok := node.(*sqlparser.Table); ok && table != nil {
 			if tableName, ok := nameMapper(table.Name.String()); ok {
+
+				tableName, enclosure, isEnclosed := getEnclosedName(tableName)
+
 				if !tableNameRegEx.MatchString(tableName) {
 					return true, &sqlparser.ErrTableNameWrongFormat{Name: tableName}
 				}
+				if isEnclosed {
+					tableName = enclosure.open + tableName + enclosure.close
+				}
 				table.Name = sqlparser.Identifier(tableName)
```

Comment:
taking the name out of the enclosure to enable the regex to validate the name, then putting it back in the enclosure if appropriate.